internal/fs/fs.go

```diff
@@ -246,10 +246,16 @@ func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHand
 
 	// When user passes allow_other flag, then other users should be able to
 	// read from cache
+	isPermissible, err := util.CheckCacheDirectoryPermissions(cacheLocation)
+	if !isPermissible {
+		panic(fmt.Sprintf("Cache Location Error: %v", err.Error()))
```

Comment:
nit: usually the error string starts with small case (and i would prefer string to be "error while creating file cache directory: %v"

Reply:
done with latest commit